### ISSUE TITLE
Concat tilize defaults to single-core

### ISSUE THREAD

**Issue by sjameelTT** [tenstorrent/tt-metal#14026]
Caught by @mbahnasTT, seems like there's an tilize call that is single core. We should either default it to multicore or determine if we can remove that conversion

**Comment by jaykru-tt** [comment]
Should we have tilize/untilize default to multicore for this? cc @sjameelTT @llongTT

**Comment by sjameelTT** [comment]
Yeah, it is set default to true on the cpp side. Does the concat call it with the use_multicore set to false?

**Comment by jungeunlim-TT** [comment]
Hi, @sjameelTT 
As I traced, the concat op calls the tilize with multi_core set to true.

**Comment by jungeunlim-TT** [comment]
@sjameelTT @amorrisonTT ,

The untilize op maps an entire tile line to a single core when memconfig is the L1 interleaved. Could this explain why it seems like that the concat op calls the untilize op with multi_core off ?

**Comment by amorrisonTT** [comment]
@jungeunlim-TT are we looking at untilize or tilize?  There is a [codepath](https://github.com/tenstorrent/tt-metal/blob/main/ttnn/cpp/ttnn/operations/data_movement/untilize_with_unpadding/device/untilize_with_unpadding_program_factory.cpp#L214) in untilize that should parallelize over height and width. I am less familiar with tilize off the top of my head.

**Comment by jungeunlim-TT** [comment]
@amorrisonTT  
I traced the tensor with interleave layout and confirmed the concat op calls both untilize and tilize with arg multicore=_true_. 

However, there seems to be a misunderstanding regarding to the untilize.

**untilize_with_unpadding_multi_core_interleaved** logic. 
 If the shape[-1] of the input is less than or equal to 1,024 (32*32), parallelized across height dim.
otherwise, parallelized across both height and width dim.

I apologize for the confusion.

**Comment by amorrisonTT** [comment]
Thank you for clarifying, that makes sense! So then the untilize height dim is probably less than 32 and only 1 core is being being used, and that would be expected behavior. If so, and there isn't some specific perf bottleneck then this might not require any additional attention
